### PR TITLE
OTWO-6946 error display on enlistment page

### PR DESCRIPTION
**Pull request by Niharika1117**

### CODE REVIEW

File: app/lib/fisbot/failure_group_api.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      :failure_groups
+    end
+
+    def failure_group_description(job_id)
+      uri = api_access.resource_uri(job_id.to_s)
+      array = JSON.parse(Net::HTTP.get(uri))
+      return array['description'] if array.present?
+    end
+  end
+end
```
Comment (notalex) [inline app/lib/fisbot/failure_group_api.rb:15]: Similar to how we have designed the `CodeLocation` model, rename this to `FailureGroup`. Rename the function at line 9 to `description`.

File: app/lib/fisbot/failure_group_api.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+class FailureGroupApi < FisbotApi
+  class << self
+    def resource
+      :failure_groups
+    end
+
+    def failure_group_description(job_id)
+      uri = api_access.resource_uri(job_id.to_s)
+      array = JSON.parse(Net::HTTP.get(uri))
+      return array['description'] if array.present?
```
Comment (notalex) [inline app/lib/fisbot/failure_group_api.rb:12]: Rename `array` to `response`. 
Once we change the route as suggested in the PR review for https://github.com/blackducksoftware/fisbot/pull/619/files, the line 10 will become:
```
uri = api_access.resource_uri("find_by_job/#{job_id}")

File: app/views/enlistments/_enlistment.html.haml [hunk 3]
```diff
@@ -2,11 +2,20 @@
...
         %span.glyphicon.glyphicon-link
+    - job_id = enlistment.fis_code_location.last_job_id
+    - fis_job = FisJob.find_by(id: job_id)
+    - description = job_id ? FailureGroupApi.failure_group_description(job_id) : ''
+    - if enlistment.code_location.do_not_fetch
+      - if description.present?
+        %p.text-danger Error: #{description}
+      - else
+        %p.text-danger= t('.dnf')
+    - elsif (enlistment.code_location.do_not_fetch == false && fis_job && fis_job.retry_count > 1 && fis_job.wait_until != nil)
```
Comment (notalex) [inline app/views/enlistments/_enlistment.html.haml:16]: Use `!enlistment.code_location.do_not_fetch`.

File: test/lib/failure_group_api_test.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+class FailureGroupApiTest < ActiveSupport::TestCase
+  it 'should get the failure groups' do
+    VCR.use_cassette('failure_groups', match_requests_on: [:path]) do
+      api = FailureGroupApi.new.fetch
+      assert JSON.parse(api)['description'] = 'No branch found'
+    end
+  end
+
+  it 'should get the failure groups for given id' do
+    VCR.use_cassette('failure_groups') do
```
Comment (notalex) [inline test/lib/failure_group_api_test.rb:14]: Name the VCR cassette by the corresponding controller + action name. In this case `failure_groups_find_by_job`.

**Comment by Niharika1117** [comment]
retest this please

**Review by notalex — APPROVED** [review]

**Pull request merged**